docs/reference-docs/auth-backend-and-frontend.md

```diff
@@ -0,0 +1,228 @@
+# Authentication and authorization
+
+WFO can be run with or without authentication. With authentication turned on authorization logic can be provided that uses - for example - user privileges to allow further access to resources.
+
+Authentication is configured using ENV variables. The frontend and backend have their own set of ENV variables and logic to be implemented to run auth(n/z).
+
+## Definitions
+
+A **frontend application** refers to a web frontend based on the frontend example ui repository: [frontend repo][1]
+A **backend application** refers to an application build using the orchestrator core as a base: [backend repo][2]
+
+## Without authentication
+
+Without authentication WFO allows all users access to all resources.
+
+##### Backend
+
+`OAUTH2_ACTIVE=false`
+
+##### Frontend:
+
+`OAUTH2_ACTIVE=false`
+
+## With Authentication
+
+WFO provides authentication based on an OIDC provider. The OIDC provider is presumed to be configured and to provide
+
+-   An authentication endpoint
+-   A tenant
+-   A client id
+-   A client secret
+
+#### Frontend
+
+The WFO frontend uses [NextAuth](3) to handle authentication. Authentication configuration can be found in [page/api/auth/[...nextauth].ts](4)
+
+**ENV variables**
+These variables need to be set for authentication to work on the frontend
+
+```
+# Auth variables
+OAUTH2_ACTIVE=true
+OAUTH2_CLIENT_ID="orchestrator-client" // The oidc client id as configured in the OIDC provider
+OAUTH2_CLIENT_SECRET=[SECRET] // The oidc client secret id as configured in the OIDC provider
+
+NEXTAUTH_PROVIDER_ID="keycloak" // String identifying the OIDC provider
+NEXTAUTH_PROVIDER_NAME="Keycloak" // The name of the OIDC provider. Keycloak uses this name to display in the login screen
+NEXTAUTH_AUTHORIZATION_SCOPE_OVERRIDE="openid profile" // Optional override of the scopes that are asked permission for from the OIDC provider
+
+# Required by the Nextauth middleware
+NEXTAUTH_URL=[DOMAIN]/api/auth // The path to the [...nextauth].js file
+NEXTAUTH_SECRET=[SECRET] // Used by NextAuth to encrypt the JWT token
+```
+
+With authentication turned on and these variables provided the frontend application will redirect unauthorized users to the login screen provided by the OIDC provider to request their credentials and return them to the page they tried to visit.
+
+Note: It's possible to add additional oidc providers including some that are provided by the NextAuth library like Google, Apple and others. See [NextAuthProviders](5) for more information.
```

Comment:
I suspect this is currently not possible due to the custom WfoProvider

Reply:
It's true that we haven't testest this. They would be in addition to the custom provder though. I have added an extra note with a disclaimer